etc/scripts/generate-github-test-data.py

```diff
@@ -131,56 +171,59 @@ def create_pull_request(repo_name, branch_name):
+        # Step 1: Create the repository
+        create_repo_url = f'{BASE_URL}/orgs/{ORGANIZATION_NAME}/repos'
+        create_repo_data = {'name': repo_name}
+        make_api_request('POST', create_repo_url, data=create_repo_data)
         print(f'Repository "{repo_name}" created successfully.')
 
-        # Step 2: Initialize the repository, add README, create workflows, and push initial commit
-        initialize_repository(repo_name)
+        try:
+            initialize_repository(repo_name)
+        except Exception as e:
+            print('ERROR:SKIPPING:REPO')
+            print(f'{e}')
+            continue
 
-        # Step 3: Create branches and commits
         for branch_index in range(args.num_branches):
+            branch_name = f'{args.issue_prefix}-{random.randint(100, 999)}-{int(time.time())}'
 
-            # Generate a unique branch name with the issue-key prefix and ranfom number
-            branch_name = f'{args.issue_prefix}-{random.randint(100, 999)}'
-            # Create a new branch
-            create_branch(branch_name)
+            # Step 4: Create a new branch
+            try:
+                create_branch(branch_name)
+            except Exception as e:
+                print('ERROR:SKIPPING:BRANCH')
+                print(f'{e}')
+                continue
 
             for commit_index in range(args.num_commits):
-                # Generate a unique file name based on epoch time and random number
-                file_name = f'file-{int(time.time())}-{random.randint(100, 999)}.txt'
-
-                # Generate a commit message based on issue prefix and random number
-                commit_message = f'{args.issue_prefix}-{random.randint(100, 999)}'
-
-                # Create and write to the file
-                with open(file_name, 'w') as f:
-                    f.write(f'Commit {commit_index + 1} on repository {repo_name}')
-
-                # Commit and push changes
-                commit_and_push_changes(branch_name, file_name, commit_message)
-
-                # Remove the file
-                os.remove(file_name)
-
-            pull_request_data = {
-                'title': f'Pull request from {branch_name} to main',
-                'head': branch_name,
-                'base': 'main'
-            }
+                file_name = f'file-{commit_index}.txt'
+                commit_message = f'{args.issue_prefix}-{random.randint(100, 999)}: Commit message'
+
+                try:
+                    # Step 6: Make a commit and push the changes
+                    commit_and_push_changes(branch_name, file_name, commit_message)
+                except Exception as e:
+                    print('ERROR:SKIPPING:COMMIT')
+                    print(f'{e}')
+                    continue
+
+            try:
+                # Step 7: Create a pull request
+                create_pull_request(repo_name, branch_name)
+            except Exception as e:
+                print('ERROR:SKIPPING:PULL')
```

Comment:
Chucked in these little logs print('ERROR:SKIPPING:PULL') just so when I watch it I can see when it happens, doesn't matter just curiosity